AppWidget/README.md

```diff
@@ -0,0 +1,331 @@
+
+AppWidget Sample
+===================================
+
+This sample demonstrates how to use the new Widgets APIs introduced in API level 31.
+The new set of APIs allows an application to build more engaging and beautiful
+widgets.
+The initial set of new APIs focuses on improving existing RemoteView APIs.
+
+List of API changes
+------------
+
+- **Widget description**
+
+  In API level 31, a widget description is shown for each widget in the widget
+  picker if `description` attribute is provided for your `appwidget-provider`
+
+  ```xml
+  <appwidget-provider
+        android:description="@string/app_widget_grocery_list_description"
+    ... />
+  ```
+
+- **Preview layout**
+
+  The widget preview is displayed from the layout XML if it's provided as
+  `previewLayout` attribute for your `appwidget-provider` instead of a
+  static drawable resource using `previewImage` attribute.
+  Note that the preview may not reflect the widget after it's actually
+  placed (e.g. colors from themes)
+
+  ```xml
+    <appwidget-provider
+        android:previewLayout="@layout/widget_grocery_list"
+    ... />
+  ```
+
+  <img src="screenshots/widget_preview.png"
+    alt="Image of the widget with description and previewLayout"
+    title="Image of the widget with description and previewLayout"
+    />
+
+- **Dynamic coloring**
+
+  In API level 31, the colors of your widget will be dynamically determined by the system colors.
+  You should use the system's default theme in order to apply the dynamic coloring (`Theme.DeviceDefault` and
+  `Theme.DeviceDefault.DayNight` for v31.
+  into account).
```

Comment:
I'm confused by what this means?

Reply:
Sorry,  a wrong sentence was added. Rephrased it.